[14.0][IMP] edi_oca: Add new model edi.configuration

thienvh332 opened the pull request.

simahawk committed:
Add edi_sale_oca

simahawk committed:
edi_sale: take over edi_sale_order_import

simahawk committed:
edi_sale_order_import: deprecate

simahawk committed:
edi_sale: filter/group by origin

simahawk committed:
edi_sale: handle ext ref for sol

simahawk committed:
edi_sale: show edi_id

simahawk committed:
edi_sale: store metadata

simahawk committed:
edi_sale: clean file and drop *type_auto dependency

simahawk committed:
edi_sale: add process hook for order create

simahawk committed:
edi_sale: process propagate edi origin

simahawk committed:
edi_sale: add edi_exchange_ready

simahawk committed:
edi_sale: fix edi disable flag

simahawk committed:
Add edi_sale_endpoint

simahawk committed:
edi_sale: fix _edi_exchange_ready for qty=0

Lines w/ no qty must not be excluded a priori

simahawk committed:
edi_sale: move OrderMixin from edi_sale_ubl

simahawk committed:
edi_sale: fix test_process

simahawk committed:
edi_sale: propagate origin on lines

simahawk committed:
edi_oca: use queue_job__no_delay key in tests

simahawk committed:
edi: fix consumer get_view w/ nested forms

Select main  only as they can be nested into field custom forms.
I'm looking at you  on v16 :S

Commit:
[MIG] edi_sale_oca: Migration to 16.0

Commit:
[IMP] edi_sale_oca: Support edi_id for edi_sale_edifact_oca

OriolMForgeFlow committed:
[IMP] edi_oca: add company_id field with multi-company rule

Commit:
[IMP] edi_sale_oca: Refactor code for passing all configurations via env_ctx

The configurations here are values for the wizard and whatever we need

Commit:
[UPD] Update edi_sale_oca.pot

OCA-git-bot committed:
[BOT] post-merge updates

Commit:
[IMP] edi_oca: support exchange filename without extension on exchange record

Commit:
[IMP] edi_oca: add partner form page

Commit:
[IMP] edi_oca: add partner form page

Commit:
[UPD] Update edi_sale_oca.pot

OCA-git-bot committed:
[BOT] post-merge updates

nilshamerlinck committed:
[FIX] edi_oca: tests using FakeModelLoader should not be post_install

tuantrantg committed:
[IMP] edi_oca: allow use of sequence for exchange filename pattern

JordiMForgeFlow committed:
[IMP] edi_oca: add encoding management for exchange type

Commit:
[FIX] edi_sale_oca: fix test

Commit:
[UPD] Update edi_sale_oca.pot

OCA-git-bot committed:
[BOT] post-merge updates

JordiMForgeFlow committed:
[IMP] edi_oca: edi_backend multi-company

Commit:
[UPD] Update edi_sale_oca.pot

gurneyalex committed:
[IMP] edi_oca: add a backend_type_code field on the backend

this field can be used in backend views to show/hide configuration elements depending on the backend type

Commit:
[IMP] edi_oca: Add Retry action on edi.exchange.record list view

mymage committed:
Added translation using Weblate (Italian)

Commit:
[IMP] edi_oca: allow to receive empty files

Currently when we receive an empty file:

- Receive step doesn't end up in state error_on_receive : that's because we just ignore the content here
- But the process step will raise an error
- So this commit goal tries to improve that
- By default, this option is disabled so we still consider that an empty file is an error case

JordiMForgeFlow committed:
[IMP] edi_oca: add button to regenerate file after error on validation

thienvh332 committed:
[BPRT] edi_sale_oca: Backport from v16.0

Commit:
[IMP] edi_oca: support modifying failed check send message

thienvh332 committed:
[IMP] edi_sale_oca: add partner form page

Commit:
[IMP] edi_oca: update docs for EDI Jobs

thienvh332 committed:
edi_oca: Add new model edi.configuration

The aim of this model is to ease configuration for all kind of exchanges
in particular at partner level.

nilshamerlinck commented on edi_oca/models/edi_backend.py:
```diff
@@ -377,6 +377,18 @@ def _cron_check_output_exchange_sync(self, **kw):
         for backend in self:
             backend._check_output_exchange_sync(**kw)
 
+    def exchange_generate_send(self, recordset, skip_send=False, skip_generate=False):
```
minor detail: skip_generate should be placed before skip_send (logical order)

nilshamerlinck commented on edi_oca/models/edi_conf_exchange_consumer_mixin.py:
```diff
@@ -0,0 +1,66 @@
+        todo = None
+        operation = "create"
+        candidates = self.browse()
+        auto_ctx = getattr(self, "_edi_auto_collect_todo", None)
```
mix with auto?

thienvh332 replied:
According to simone's description, it seems to me that he wants the user to be able to change "Todo" (**EDIAutoInfo**) and **execute it**. So I check if _auto is installed to get "Todo". That's my idea

QuocDuong1306 replied:
How about this one a Thien?

![image](https://github.com/user-attachments/assets/6c103bd1-53e4-4d8e-b6a0-582fc3cc259e)

As I know, `*_auto` has been implemented in `create() / write()` as well, no?
Assume that, `*_auto` is installed into the same DB with `edi.configuration`, if we took into account `auto` flows in here, It would run twice in total, right? (1st for `*_auto`, 2nd for here)

nilshamerlinck replied:
correct, the goal is to completely get rid of the `*_auto` module

nilshamerlinck commented on edi_oca/models/edi_configuration.py:
```diff
@@ -0,0 +1,193 @@
+    trigger = fields.Selection(
+        [
+            ("none", "None"),
+            ("on_post_account_move", "On Post Account Move"),
```
these triggers values should be added by their respective modules
e.g. `on_button_confirm_purchase_order`: `edi_purchase_oca`

nilshamerlinck replied:
no?

thienvh332 replied:
I just think you haven't finished reviewing it yet so I haven't edited it yet. I will fix it

nilshamerlinck commented on edi_oca/models/edi_exchange_consumer_mixin.py:
```diff
@@ -291,3 +291,38 @@ def _edi_set_origin(self, exc_record):
+        exchange_record._trigger_edi_event("generated")
+
+    def _edi_send_via_email(self, ir_action=None, partners=None):
+        if ir_action is None:
+            if self._name == "purchase.order":
+                ir_action = self.action_rfq_send()
+            elif self._name == "sale.order":
+                ir_action = self.action_quotation_send()
+            else:
+                return False
+
+        ctx = ir_action.get("context", {})
+        composer_model = self.env[ir_action["res_model"]].with_context(ctx)
+
+        if self._name == "purchase.order":
+            subtype = self.env.ref("purchase.mt_rfq_approved")
+            composer = composer_model.create({"subtype_id": subtype.id})
+            composer.onchange_template_id_wrapper()
+            composer.partner_ids = self._vendor_email_partners().ids
+        elif self._name == "sale.order":
+            subtype = self.env.ref("sale.mt_order_confirmed")
+            composer = composer_model.create({"subtype_id": subtype.id})
+            composer.onchange_template_id_wrapper()
+            composer.partner_ids = self.partner_id.ids
+        else:
```
belongs in their respective modules no? edi_purchase_oca/edi_sale_oca

thienvh332 replied:
Yes, I moved it from cosaflow_edi_purchase. And fix it a little.

QuocDuong1306 replied:
After moving the code,
I think we can make the function smarter: `gettatr(self, f"_edi_send_{self._name}_via_email")(...)`

QuocDuong1306 committed:
[IMP] edi_oca: add btn to show all related queue jobs

alexis-via committed:
[FIX] account_invoice_import_simple_pdf: version extraction for fitz/PyMuPDF

Use with statement for NamedTemporaryFile()

alexis-via committed:
[FIX] account_invoice_import_simple_pdf: add seek(0) to avoid error "empty file"

Add try/except on pypdf text extraction

MiquelRForgeFlow committed:
[FIX] edi_oca: improve error in order to be helpful

QuocDuong1306 commented on edi_oca/data/edi_configuration.xml:
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
```
`noupdate` will be better, right?

QuocDuong1306 commented on edi_oca/models/edi_backend.py:
```diff
@@ -377,6 +377,18 @@ def _cron_check_output_exchange_sync(self, **kw):
+                # Chain send job.
+                # Raise prio to max to send the record out as fast as possible.
+                job1.on_done(rec.delayable(priority=0).action_exchange_send())
+            job1.delay()
```
default for `skip_send` should be True

QuocDuong1306 commented on edi_oca/models/edi_backend.py:
```diff
@@ -396,13 +408,8 @@ def _check_output_exchange_sync(
-                job1.on_done(rec.delayable(priority=0).action_exchange_send())
-            job1.delay()
+        if new_records:
+            self.exchange_generate_send(new_records, skip_send)
```
```suggestion
            self.exchange_generate_send(new_records, skip_send=skip_send)
```
Make sure overriding `skip_send` instead of `skip_generate`

QuocDuong1306 commented on edi_oca/models/edi_configuration.py:
```diff
@@ -0,0 +1,189 @@
+        model_field="model",
+        copy=False,
+        help="",
+    )
```
You can reference to [here](https://github.com/OCA/edi/blob/14.0/edi_oca/models/edi_exchange_record.py) to init the suitable attributes for fields (eg: `name` should be required, define `record` property, display_name,...)

QuocDuong1306 commented on edi_oca/models/edi_configuration.py:
```diff
@@ -0,0 +1,189 @@
+            ("on_email_send", "Send Email"),
+        ],
+        string="Trigger",
+        default="none",
```
sorry, but do we need the `none` value? how about `default=False`

QuocDuong1306 commented on edi_oca/models/edi_configuration.py:
```diff
@@ -0,0 +1,189 @@
+
+    # @api.depends("model_id", "res_id")
+    # def _compute_record_id(self):
+    #     pass
```
Something like: https://github.com/OCA/edi/blob/e9fe1a70f7db810dfd5da879b8f25a15f490951a/edi_oca/models/edi_exchange_record.py#L205

QuocDuong1306 commented on edi_oca/models/edi_configuration.py:
```diff
@@ -0,0 +1,189 @@
+            "datetime": safe_eval.datetime,
+            "dateutil": safe_eval.dateutil,
+            "time": safe_eval.time,
+            "utc_now": datetime.datetime.utcnow().isoformat(),
```
Can we use `fields.Datetime.now()`?

QuocDuong1306 commented on edi_oca/models/edi_configuration.py:
```diff
@@ -0,0 +1,189 @@
+
+    def edi_exec_snippet_do(self, record, **kwargs):
+        self.ensure_one()
+        old_value = kwargs.get("old_vals", {}).get(record.id, False)
+        new_value = kwargs.get("vals", {}).get(record.id, False)
```
```suggestion
        old_value = kwargs.get("old_vals", {}).get(record.id, {})
        new_value = kwargs.get("vals", {}).get(record.id, {})
```
In some cases, we want to check the key in dict, it will support for `in` operator

QuocDuong1306 commented on edi_oca/models/edi_configuration.py:
```diff
@@ -0,0 +1,189 @@
+        ctx.update(self._time_utils())
+        return ctx
+
+    def _evaluate_code_snippet(self, snippet, **render_values):
+        if not self._code_snippet_valued(snippet):
+            return {}
+        eval_ctx = dict(render_values, **self._get_code_snippet_eval_context())
+        safe_eval.safe_eval(snippet, eval_ctx, mode="exec", nocopy=True)
+        result = eval_ctx.get("result", {})
+        if not isinstance(result, dict):
+            return {}
+        return result
```
Actually, I didn't see a `record` variable in the code (for `eval_ctx`)

QuocDuong1306 commented on edi_oca/models/edi_configuration.py:
```diff
@@ -0,0 +1,189 @@
+            record._edi_auto_handle(vals["todo"])
+        self._evaluate_code_snippet(self.snippet_do, **vals)
+
+    @api.model
+    def edi_get_conf(self, trigger, backend=None):
+        domain = [("trigger", "=", trigger)]
+        if backend:
+            domain.append(("backend_id", "=", backend.id))
+        else:
+            domain.append(("backend_id", "=", self.type_id.backend_id.id))
+        return self.search(domain)
```
Missing `active` ?
Could we make this search smatter with `model` and `type` (optional)?

QuocDuong1306 commented on edi_oca/models/edi_exchange_record.py:
```diff
@@ -308,6 +308,9 @@ def action_exchange_generate(self, **kw):
         self.ensure_one()
         return self.backend_id.exchange_generate(self, **kw)
 
+    def action_exchange_generate_send(self):
+        return self.backend_id.exchange_generate_send(self)
```
missing `self.ensure_one()`?

QuocDuong1306 commented on edi_oca/views/edi_configuration_views.xml:
```diff
@@ -0,0 +1,98 @@
+            </form>
+        </field>
+    </record>
+    <record model="ir.actions.act_window" id="act_open_edi_configuration_view">
+        <field name="name">EDI Configuration</field>
+        <field name="type">ir.actions.act_window</field>
+        <field name="res_model">edi.configuration</field>
+        <field name="view_mode">tree,form</field>
+        <field name="domain">[]</field>
+    </record>
```
We can make the default search for `active` configurations

QuocDuong1306 commented on edi_purchase_oca/components/listener_purchase_order.py:
```diff
@@ -0,0 +1,38 @@
+        for rec in record:
+            confs = rec.partner_id.edi_purchase_conf_ids.edi_get_conf(trigger)
+            for conf in confs:
+                if conf.model == rec._name:
+                    conf.edi_exec_snippet_do(rec)
```
Make `edi_get_conf` smarter with `model` and `type`...

QuocDuong1306 commented on edi_purchase_oca/data/edi_configuration.xml:
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
```
missing `noupdate`?

QuocDuong1306 commented on edi_oca/models/edi_configuration.py:
```diff
@@ -0,0 +1,189 @@
+    model = fields.Many2one(
+        "ir.model", string="Model", help="model the conf applies to"
+    )
+    res_id = fields.Many2one(
+        "res.partner", string="Partner", ondelete="cascade", required=False
+    )
```
I think we have some mistakes, why `res_id` here only references to `res.partner`?

I can't get a whole context for a single record here, maybe will discuss later

QuocDuong1306 commented on edi_oca/views/edi_configuration_views.xml:
```diff
@@ -0,0 +1,98 @@
+                        <page name="snippet_do" string="EDI Config Do">
+                            <field name="snippet_do" widget="ace" colspan="4" />
+                        </page>
+                        <page name="res_id" string="Partners">
+                          <field name="res_id" />
+                        </page>
```
Waiting for discussing...

QuocDuong1306 commented on edi_purchase_oca/models/res_partner.py:
```diff
@@ -0,0 +1,15 @@
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    edi_purchase_conf_ids = fields.One2many(
+        comodel_name="edi.configuration",
+        inverse_name="res_id",
+        string="EDI Purchase Config Ids",
+    )
```
missing?
```
add res.partner.edi_purchase_conf_ids
domain model=purchase.order, default_model=purchase.order, default_res_id=active_id

nilshamerlinck commented on edi_purchase_oca/tests/test_edi_configuration.py:
```diff
@@ -0,0 +1,85 @@
+                "code": "create_config",
+                "backend_id": self.backend.id,
+                "type_id": self.exchange_type_out.id,
+                "trigger": "on_record_create",
```
on_button_confirm_purchase_order is probably more significant to test in that module

nilshamerlinck commented on edi_purchase_oca/tests/test_edi_configuration.py:
```diff
@@ -0,0 +1,85 @@
+# Copyright 2024 CamptoCamp SA,
```
```suggestion
# Copyright 2024 CamptoCamp SA
```
```suggestion
# Copyright 2024 CamptoCamp SA,
```

nilshamerlinck replied:
```suggestion
# Copyright 2024 CamptoCamp SA
```

nilshamerlinck commented on edi_oca/models/edi_exchange_consumer_mixin.py:
```diff
@@ -291,3 +291,38 @@ def _edi_set_origin(self, exc_record):
+        exchange_record._notify_related_record(msg)
+        exchange_record._trigger_edi_event("generated")
+
+    def _edi_send_via_email(self, ir_action=None, partners=None):
+        if ir_action is None:
+            if self._name == "purchase.order":
+                ir_action = self.action_rfq_send()
+            elif self._name == "sale.order":
+                ir_action = self.action_quotation_send()
+            else:
+                return False
+
+        ctx = ir_action.get("context", {})
+        composer_model = self.env[ir_action["res_model"]].with_context(ctx)
+
+        if self._name == "purchase.order":
+            subtype = self.env.ref("purchase.mt_rfq_approved")
+            composer = composer_model.create({"subtype_id": subtype.id})
+            composer.onchange_template_id_wrapper()
+            composer.partner_ids = self._vendor_email_partners().ids
+        elif self._name == "sale.order":
+            subtype = self.env.ref("sale.mt_order_confirmed")
+            composer = composer_model.create({"subtype_id": subtype.id})
+            composer.onchange_template_id_wrapper()
+            composer.partner_ids = self.partner_id.ids
```
this should be generalized and specific part for sale/purchase done in snippets in their respective modules (see specs)

nilshamerlinck replied:
need to call a hook _get_composer that will be defined in their respective modules

nilshamerlinck commented on edi_purchase_oca/components/listener_purchase_order.py:
```diff
@@ -0,0 +1,41 @@
+    _inherit = "base.event.listener"
+    _apply_on = ["purchase.order"]
+
+    def on_button_confirm_purchase_order(self, record):
+        trigger = "on_button_confirm_purchase_order"
+        for rec in record:
+            confs = self.env["edi.configuration"].edi_get_conf(
+                trigger, rec._name, rec.partner_id
+            )
+            for conf in confs:
+                conf.edi_exec_snippet_do(rec)
+
+    def on_record_write(self, record, fields=None, **kwargs):
+        trigger = "on_record_write"
+        if kwargs.get("vals", False):
+            for rec in record:
+                confs = self.env["edi.configuration"].edi_get_conf(
+                    trigger, rec._name, rec.partner_id
+                )
+                for conf in confs:
+                    conf.edi_exec_snippet_do(rec, **kwargs)
```
these ones are not for `purchase.order` only, right? what about creating a `edi.config.consumer.listener` in `edi_oca`? then EDIConfigPurchaseListener can inherit from it

nilshamerlinck commented on edi_purchase_oca/components/listener_purchase_order.py:
```diff
@@ -0,0 +1,41 @@
+            for conf in confs:
+                conf.edi_exec_snippet_do(rec)
+
+    def on_record_write(self, record, fields=None, **kwargs):
+        trigger = "on_record_write"
+        if kwargs.get("vals", False):
+            for rec in record:
+                confs = self.env["edi.configuration"].edi_get_conf(
+                    trigger, rec._name, rec.partner_id
+                )
+                for conf in confs:
+                    conf.edi_exec_snippet_do(rec, **kwargs)
+
+    def on_record_create(self, record, fields=None, **kwargs):
+        trigger = "on_record_create"
+        val_list = kwargs.get("vals", False)
+        if val_list:
+            for rec, vals in zip(record, val_list):
+                kwargs["vals"] = {rec.id: vals}
+                confs = self.env["edi.configuration"].edi_get_conf(
+                    trigger, rec._name, rec.partner_id
+                )
+                for conf in confs:
+                    conf.edi_exec_snippet_do(rec, **kwargs)
```
these ones are not for `purchase.order` only, right? what about creating a `edi.config.consumer.listener` in `edi_oca`? then EDIConfigPurchaseListener can inherit from it

nilshamerlinck commented on edi_oca/models/edi_exchange_consumer_mixin.py:
```diff
@@ -291,3 +291,38 @@ def _edi_set_origin(self, exc_record):
+
+    def _edi_send_via_email(self, ir_action=None, partners=None):
+        if ir_action is None:
+            if self._name == "purchase.order":
+                ir_action = self.action_rfq_send()
+            elif self._name == "sale.order":
+                ir_action = self.action_quotation_send()
```
can be removed because you have it in snippets

nilshamerlinck commented on edi_oca/models/edi_configuration.py:
```diff
@@ -0,0 +1,217 @@
+    )
+
+    @api.constrains("backend_id", "type_id")
+    def _constrain_backend(self):
```
```suggestion
    def _constrains_backend(self):
```

nilshamerlinck commented on edi_purchase_oca/models/edi_configuration.py:
```diff
@@ -0,0 +1,15 @@
+
+
+class EdiConfiguration(models.Model):
+    _inherit = "edi.configuration"
```
_description

nilshamerlinck commented on edi_oca/views/menuitems.xml:
```diff
@@ -82,4 +82,11 @@
+    <menuitem
+        id="menu_edi_configuration"
+        parent="edi_oca.menu_edi_config"
+        name="EDI Configuration"
```
```suggestion
        name="Configurations"
```

nilshamerlinck commented on edi_purchase_oca/views/res_partner_view.xml:
```diff
@@ -0,0 +1,18 @@
+        <field name="model">res.partner</field>
+        <field name="inherit_id" ref="base.view_partner_form" />
+        <field name="arch" type="xml">
+            <xpath expr="//page[@name='sales_purchases']" position="inside">
```
better in its own tab EDI no?

thienvh332 replied:
Because it is in the depend relationship of **`edi_purchase_oca`**. The **`EDI tab`** is not created yet. So I will create for it an **`EDI Configurations`** tab to contain the relationships between **partner**, **model**, **edi.configuration**

thienvh332 committed:
[FIX] pdf_helper: fix issue pypdf 5.0 dropped support for python<=3.7

alexis-via committed:
[FIX] account_invoice_download: don't set default_name in context

If default_name is set in context just to have a default label on the
account.invoice.import.download, it will also be used if do a manual run
right after, so you will get draft invoices with a 'name' already set !

OCA-git-bot committed:
Merge PR #1041 into 14.0

Signed-off-by simahawk

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #1018 into 14.0

Signed-off-by alexis-via

OCA-git-bot committed:
[BOT] post-merge updates

HviorForgeFlow committed:
[IMP] edi_oca: Add hook on new output exchange record search

alexis-via committed:
[FIX] pypdf version for tests

OCA-git-bot committed:
Merge PR #1043 into 14.0

Signed-off-by alexis-via

OCA-git-bot committed:
[BOT] post-merge updates

simahawk committed:
Update .oca/oca-port/blacklist/edi_oca.json

OCA-git-bot committed:
Merge PR #1023 into 14.0

Signed-off-by simahawk

Commit:
[UPD] Update edi_oca.pot

Commit:
[UPD] Update edi_sale_oca.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-14.0/edi-14.0-edi_oca
Translate-URL: https://translation.odoo-community.org/projects/edi-14-0/edi-14-0-edi_oca/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-14.0/edi-14.0-edi_sale_oca
Translate-URL: https://translation.odoo-community.org/projects/edi-14-0/edi-14-0-edi_sale_oca/

OCA-git-bot committed:
Merge PR #1066 into 14.0

Signed-off-by simahawk

Commit:
[UPD] Update edi_sale_order_import.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-14.0/edi-14.0-edi_sale_order_import
Translate-URL: https://translation.odoo-community.org/projects/edi-14-0/edi-14-0-edi_sale_order_import/

OCA-git-bot committed:
Merge PR #1022 into 14.0

Signed-off-by simahawk

Commit:
[UPD] Update edi_sale_endpoint.pot

OCA-git-bot committed:
[BOT] post-merge updates

simahawk committed:
edi_sale_oca: use specific demo data for tests

simahawk committed:
edi_oca: edi.conf cleanup tests

simahawk committed:
edi_oca: add edi.configuration.trigger

Ease configuration by filtering triggers by model.
In the long run we'll have tons of triggers for specific scopes (sale, purchase, etc).
This change will make sure the selection of the trigger won't be cluttered.
It also adds the possibility to describe triggers as users prefer.

simahawk committed:
edi_oca: consumer mixin add helper methods to send via edi

simahawk committed:
edi_sale_oca: handle existing order drop dead code

This piece of code was never executed because it should have been done
with a new cursor. Yet, there's no need to post a message on the order.
Another point is: shall we simply avoid failing and return the msg to the job result?

OCA-git-bot committed:
Merge PR #1052 into 14.0

Signed-off-by simahawk

Commit:
[UPD] Update edi_oca.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-14.0/edi-14.0-edi_oca
Translate-URL: https://translation.odoo-community.org/projects/edi-14-0/edi-14-0-edi_oca/

simahawk committed:
edi_sale_oca: improve test mixin

simahawk committed:
Update .oca/oca-port/blacklist/edi_oca.json

simahawk committed:
edi_oca: fix CONFIGURE.rst syntax

simahawk committed:
edi_oca: adapt tests for v14

OCA-git-bot committed:
Merge PR #1029 into 14.0

Signed-off-by simahawk

Commit:
[UPD] Update edi_oca.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-14.0/edi-14.0-edi_oca
Translate-URL: https://translation.odoo-community.org/projects/edi-14-0/edi-14-0-edi_oca/

simahawk committed:
edi_oca: consumer mixin trigger state event

Models using the consumer mixing and having a state field
will now trigger a specific event when the state is updated.

simahawk committed:
edi_sale_oca: drop dead job function

Was used only when plugged w/ edi_exchange_type_auto in a prototype.

simahawk committed:
edi_sale_oca: split input to edi_sale_input_oca

simahawk committed:
edi_sale_order_import: update deprecate msg

simahawk committed:
edi_sale_oca: 14.0.2.0.0

OCA-git-bot committed:
Merge PR #1075 into 14.0

Signed-off-by simahawk

Commit:
[UPD] Update edi_sale_input_oca.pot

Commit:
[UPD] Update edi_sale_oca.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-14.0/edi-14.0-edi_sale_oca
Translate-URL: https://translation.odoo-community.org/projects/edi-14-0/edi-14-0-edi_sale_oca/

simahawk committed:
edi_oca: refactor exchange_generate_send

Make it easier to understand how it works and avoid checks on attributes.

simahawk committed:
edi_oca: edi_conf avoid clash of label for model fields
